crypto/evp/keymgmt_lib.c

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "internal/cryptlib.h"
+#include "internal/nelem.h"
+#include "internal/evp_int.h"
+#include "internal/asn1_int.h"
+#include "internal/provider.h"
+#include "evp_locl.h"
+
+static OSSL_PARAM *paramdefs_to_params(const OSSL_PARAM *paramdefs)
+{
+    size_t cnt;
+    const OSSL_PARAM *p;
+    OSSL_PARAM *params, *q;
+
+    for (cnt = 0, p = paramdefs; p->key != NULL; p++, cnt++)
```

Comment:
This looks off by one low?

Reply:
I don't follow...

Reply:
Consider, the trivial case: `paramdefs = { OSSL_PARAM_END }`.
After the loop finishes, `cnt` will be zero.

The allocated array (of zero length) would not have enough space to be terminated but it is terminated.

Changing the `cnt = 0` to `cnt = 1` should fix this.

Reply:
Okie, will look again

Reply:
Fixed